lib/modules.nix

```diff
@@ -93,7 +93,10 @@ rec {
             res set._definedNames
         else
           res;
-      result = { inherit options config; };
+      result = {
+        inherit options;
+        config = removeAttrs config [ "_module" ];
```

Comment:
This wouldn't be directly accessable for NixOS, since `nixos/lib/eval-config.nix` and `nixos/default.nix` don't expose that attribute. I guess exposing it like this wouldn't be a problem though.

Reply:
Yes, some glue code may need extending, but at least it's a starting point.